api/v1beta1/verticadb_types.go

```diff
@@ -1084,6 +1084,12 @@ type SubclusterStatus struct {
 	// Object ID of the subcluster.
 	Oid string `json:"oid"`
 
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	// +optional
+	// The type of the subcluster. It could be primary, secondary, transient, or empty if
```

Comment:
I don't think transient should be part of this list; it is not real subcluster type. In the database a transient sc would be secondary so that's the type that should appear here.
Also add other types like sandboxprimary/sandboxsecondary.